- added login functionality in userService

SamuelBruegger opened the pull request:
- added test case for login in UserServiceIntegrationTest
- added test case for login in UserServiceTest
- changed SchedulingSession, because there was a problem with the Lists
- UserControllerTest in branch User-Controller-Tests

closes #22

SamuelBruegger committed:
- added login functionality in userService
- added test case for login in UserServiceIntegrationTest
- added test case for login in UserServiceTest
- changed SchedulingSession, because there was a problem with the Lists
- UserControllerTest in branch User-Controller_Tests

SamuelBruegger committed:
- test were in comment (because not every function used for the tests is implemented yet)

AleksandarRistic99 commented on src/main/java/ch/uzh/ifi/hase/soprafs21/service/UserService.java:
```diff
@@ -39,4 +46,39 @@ public User createUser(User newUser) {
+     * @throws org.springframework.web.server.ResponseStatusException
+     * @see User
+     */
+    private void checkIfUserExists(User userToCheck) {
```
since I have implemeted this method also, it may be good to delete it here to avoid merge conflicts ?

Taremeh replied:
Would be possible, however there will probably be quite a few merge conflicts (also with #67) because of incomplete implementation of the `User` class as well...

AleksandarRistic99 commented:
De rest sett aber denki guet sii

AleksandarRistic99 commented on src/main/java/ch/uzh/ifi/hase/soprafs21/service/UserService.java:
```diff
@@ -39,4 +46,39 @@ public User createUser(User newUser) {
+    }
+
+    public String loginUser(UserPostDTO userPostDTO) {
+        User userByUsername = userRepository.findByEmail(userPostDTO.getEmail());
```
i think this should be called User userByEmail instead of UserByUsername

Taremeh commented on src/main/java/ch/uzh/ifi/hase/soprafs21/controller/UserController.java:
```diff
@@ -48,11 +48,7 @@ public UserGetDTO createUser(@RequestBody UserPostDTO userPostDTO) {
-
-        // checks User input and returns token when successful
-        //return userService.loginUser(userPostDTO);
+        return userService.loginUser(userPostDTO);
```
⚠️ userPostDTO needs to be converted to internal representation (namely type `User`) before passing it to the user service. (Via DTOMapper)

Taremeh reviewed (changes requested):
See inline comments.

For consistency reasons: Please make getters & setters one-liners (e.g. UserGetDTO, UserPostDTO ...).

Just out of curiosity: Why are the Lists being changed to Sets (e.g. in SchedulingSession.java)? Fine if they don't have to be ordered.

SamuelBruegger committed:
Merge branch 'master' into login-functionality

# Conflicts:
#	src/main/java/ch/uzh/ifi/hase/soprafs21/service/UserService.java

SamuelBruegger committed:
Merge branch 'master' into login-functionality

# Conflicts:
#	src/main/java/ch/uzh/ifi/hase/soprafs21/entities/SchedulingSession.java
#	src/main/java/ch/uzh/ifi/hase/soprafs21/entities/User.java
#	src/main/java/ch/uzh/ifi/hase/soprafs21/rest/dto/UserPostDTO.java
#	src/test/java/ch/uzh/ifi/hase/soprafs21/service/UserServiceIntegrationTest.java
#	src/test/java/ch/uzh/ifi/hase/soprafs21/service/UserServiceTest.java

SamuelBruegger committed:
- changed login functionality to get a user parameter as input
- changed name of user from UserByUsername to UserByEmail
- checked that all tests pass

SamuelBruegger committed:
- removed notBlank annotations from bio, name and gender to solve a problem with the DTO-mapper

SamuelBruegger committed:
- removed bio, name and gender from DTO mapper

SamuelBruegger committed:
- resolved DTOMappterTest with new PostDTOMapper

SamuelBruegger committed:
- resolved UserServiceIntegrationTest with new PostDTOMapper

AleksandarRistic99 commented on src/test/java/ch/uzh/ifi/hase/soprafs21/service/UserServiceIntegrationTest.java:
```diff
@@ -95,19 +88,13 @@ public void createUser_logOutUser_success(){
-        user.setPassword("testPassword");
-        user.setGender(Gender.MALE);
-        user.setBio("asdf");
+        testUser.setPassword("testPassword2");
```
das muess do glaubi user.setPassword sii oder?

AleksandarRistic99 commented on src/main/java/ch/uzh/ifi/hase/soprafs21/rest/mapper/DTOMapper.java:
```diff
@@ -21,9 +21,6 @@
 
     @Mapping(source = "email", target = "email")
     @Mapping(source = "password", target = "password")
-    @Mapping(source = "name", target = "name")
```
und name muesch dinne loh, wel de name chont be de registrierig

AleksandarRistic99 commented on src/test/java/ch/uzh/ifi/hase/soprafs21/rest/mapper/DTOMapperTest.java:
```diff
@@ -25,19 +25,13 @@ public void testCreateUser_fromUserPostDTO_toUser_success() {
         UserPostDTO userPostDTO = new UserPostDTO();
         userPostDTO.setEmail("name@uzh.ch");
         userPostDTO.setPassword("blabla");
-        userPostDTO.setName("name");
```
be de tests au bitte de name dinne loh

AleksandarRistic99 commented on src/test/java/ch/uzh/ifi/hase/soprafs21/rest/mapper/DTOMapperTest.java:
```diff
@@ -25,19 +25,13 @@ public void testCreateUser_fromUserPostDTO_toUser_success() {
         // check content
         assertEquals(userPostDTO.getEmail(), user.getEmail());
         assertEquals(userPostDTO.getPassword(), user.getPassword());
-        assertEquals(userPostDTO.getName(), user.getName());
```
same here

AleksandarRistic99 commented on src/test/java/ch/uzh/ifi/hase/soprafs21/service/UserServiceIntegrationTest.java:
```diff
@@ -47,10 +48,7 @@ public void createUser_validInputs_success() {
 
         User testUser = new User();
         testUser.setEmail("test.user@uzh.ch");
-        testUser.setName("Tester");
```
same here

AleksandarRistic99 commented on src/test/java/ch/uzh/ifi/hase/soprafs21/service/UserServiceIntegrationTest.java:
```diff
@@ -70,10 +68,7 @@ public void createUser_duplicateEmail_throwsException() {
 
         User testUser = new User();
         testUser.setEmail("test.user@uzh.ch");
-        testUser.setName("Tester");
```
same here

AleksandarRistic99 commented on src/test/java/ch/uzh/ifi/hase/soprafs21/service/UserServiceIntegrationTest.java:
```diff
@@ -95,19 +88,13 @@ public void createUser_logOutUser_success(){
 
         User testUser = new User();
         testUser.setEmail("test.user@uzh.ch");
-        testUser.setName("Tester");
```
same here

AleksandarRistic99 commented on src/test/java/ch/uzh/ifi/hase/soprafs21/service/UserServiceIntegrationTest.java:
```diff
@@ -95,19 +88,13 @@ public void createUser_logOutUser_success(){
         // reset LocalDateTime and token
         User user = new User();
         user.setEmail("test2.user@uzh.ch");
-        user.setName("Tester2");
```
same here

SamuelBruegger committed:
- added parameter name back into PostDTOMapper
